### PR TITLE
Add IntelliJ support to supported extensions

### PR DESCRIPTION
**Pull request by milanwilliams**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

### CODE REVIEW

**Review by s-santillan — APPROVED** [review]
This is pretty great!

File: docs/extensions/semgrep-intellij.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+# To try Semgrep without installation run via Docker
+$ docker run --rm -v "${PWD}:/src" returntocorp/semgrep semgrep
+```
+:::
+
+## Quick start
+
+1. Install the Semgrep extension:
+   -  Visit [Semgrep's page on the Jet Brains Marketplace](https://plugins.jetbrains.com/plugin/22622-semgrep).
```
Comment (neo413001) [inline docs/extensions/semgrep-intellij.md:31]: JetBrains instead of Jet Brains

File: docs/extensions/semgrep-intellij.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+Sign in to get access to Semgrep Code, Supply Chain, and Secrets. Scan your code and package dependencies for known issues, software vulnerabilities, and detected secrets. Get inline results and problem highlighting.
+
+### Automatic scanning
+
+When you open a file, Semgrep scans it right away.
+
+### Rule Quick Links
+
+Want to go to the definition of a rule? Hover over a match and click the link!
```
Comment (neo413001) [inline docs/extensions/semgrep-intellij.md:98]: this feels like a marketing message but if @s-santillan is okay with it, I am okay too :) 
ALT: 
Hover over a match and click the link.

File: docs/extensions/semgrep-intellij.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+:::tip
+You can also click the Semgrep icon in the IntelliJ toolbar to quickly access all available commands.
+:::
+
+## Features
+
+### Scan with Semgrep Code, Supply Chain, and Secrets 
+
+Sign in to get access to Semgrep Code, Supply Chain, and Secrets. Scan your code and package dependencies for known issues, software vulnerabilities, and detected secrets. Get inline results and problem highlighting.
```
Comment (neo413001) [inline docs/extensions/semgrep-intellij.md:90]: Does it find Secrets too?

File: docs/extensions/semgrep-intellij.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+2. Sign in: Press <kbd>Ctrl+⇧Shift+A</kbd> (Windows) or <kbd>⌘Command+⇧Shift+A</kbd> (macOS) and sign in to Semgrep Cloud Platform by selecting the following command:
+   ```
+   Sign in with Semgrep
+   ```
+3. Test the extension by pressing <kbd>Ctrl+⇧Shift+A</kbd> (Windows) or <kbd>⌘Command+⇧Shift+A</kbd> (macOS) and run the following command:
+   ```
+   Scan workspace with Semgrep
+   ```
+4. See Semgrep findings: Hold the pointer over the code that has the red underline.
```
Comment (neo413001) [inline docs/extensions/semgrep-intellij.md:42]: Does the "hover over a line to see more information about the finding" work for IntelliJ?

**Review by neo413001 — CHANGES_REQUESTED** [review]

File: docs/extensions/semgrep-intellij.md (hunk 2 above)
Comment (s-santillan) [reply docs/extensions/semgrep-intellij.md:98]: Good catch, let's stick with the ALT that Chinmay suggested.

File: docs/extensions/semgrep-intellij.md (hunk 4 above)
Comment (s-santillan) [reply docs/extensions/semgrep-intellij.md:42]: @milanwilliams is SME for this, so I'll defer to her, but I assumed so as she wrote it.

File: docs/extensions/semgrep-intellij.md (hunk 3 above)
Comment (ajbt200128) [reply docs/extensions/semgrep-intellij.md:90]: No, the extension doesn't support supply chain, secrets, or pro rn

File: docs/extensions/semgrep-intellij.md (hunk 4 above)
Comment (ajbt200128) [reply docs/extensions/semgrep-intellij.md:42]: yea it should, line can also be different colors (highlighted yellow in intellij)

File: docs/extensions/semgrep-intellij.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+```sh
+# For macOS
+$ brew install semgrep
+
+# For Ubuntu/WSL/Linux/macOS
+$ python3 -m pip install semgrep
+
+# To try Semgrep without installation run via Docker
+$ docker run --rm -v "${PWD}:/src" returntocorp/semgrep semgrep
```
Comment (ajbt200128) [inline docs/extensions/semgrep-intellij.md:24]: Not sure if this is relevant for the extension

File: docs/extensions/semgrep-intellij.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
+---
+slug: semgrep-intellij
+append_help_link: true
+---
+
+import MoreHelp from "/src/components/MoreHelp"
+
+# Semgrep IntelliJ extension
+
+[Semgrep](https://semgrep.dev/) accelerates your security journey by swiftly scanning code and package dependencies for known issues, software vulnerabilities, and detected secrets with unparalleled efficiency. By default, the Semgrep IntelliJ extension scans code whenever you change or open files.
```
Comment (s-santillan) [inline docs/extensions/semgrep-intellij.md:10]:
```suggestion
[Semgrep](https://semgrep.dev/) swiftly scans code and package dependencies for known issues, software vulnerabilities, and detected secrets. Run Semgrep in your developer environment with the IntelliJ extension to catch code issues as you type. By default, the Semgrep IntelliJ extension scans code whenever you change or open files.
```
As per Chinmay's observation further down the doc, I've suggested some trims to the more markety bits and focused on the functionality.

**Review by s-santillan — APPROVED** [review]

File: docs/extensions/semgrep-intellij.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+IntelliJ extension does not support: 
+- IntelliJ IDEA Community Edition
+
+:::
+
+## Commands
+
+Run Semgrep extension commands through the IntelliJ Command Palette. You can access the Command Palette by pressing <kbd>Ctrl+⇧Shift+A</kbd> (Windows) or <kbd>⌘Command+⇧Shift+A</kbd> (macOS) on your keyboard. 
+
+- `Sign in with Semgrep`: Sign up or log in to the Semgrep Cloud Platform (this command opens a new window in your browser). When you sign up, you can automatically scan with Semgrep [Code](https://semgrep.dev/products/semgrep-code/), [Supply Chain](https://semgrep.dev/products/semgrep-supply-chain/), and [Secrets](https://semgrep.dev/products/semgrep-secrets/). Alternatively, you can log in through your command-line interface by running `semgrep login`.
```
Comment (s-santillan) [inline docs/extensions/semgrep-intellij.md:74]:
```suggestion
- `Sign in with Semgrep`: Sign up or log in to the Semgrep Cloud Platform (this command opens a new window in your browser). Alternatively, you can log in through your command-line interface by running `semgrep login`.
```
There is a slight implication that the extension scans for those products, including Supply Chain and Secrets, when it does not. Let's just focus on the extension's capabilities for now.

**Comment by s-santillan** [comment]
@neo413001 your previous review is going to block @milanwilliams from merging - can you approve on your end?

**Pull request merged**